src/doc/rustc/src/check-cfg/cargo-specifics.md

```diff
@@ -0,0 +1,96 @@
+# Cargo specifics - Checking conditional configurations
```

Comment:
When I suggested this go into the check-cfg docs, it was with the assumption that it was to be an additional note that built on existing knowledge.  I also didn't realize there was a separate section for this and thought it would be in [the lint docs](https://doc.rust-lang.org/nightly/rustc/lints/listing/warn-by-default.html#unexpected-cfgs).

Personally, what I recommend is
- Update the [lint docs](https://doc.rust-lang.org/nightly/rustc/lints/listing/warn-by-default.html#unexpected-cfgs) to point to the [check-cfg docs](https://doc.rust-lang.org/nightly/rustc/check-cfg.html) so users can discover how to customize things
- Soon after `It has this basic form:` in the [check-cfg docs](https://doc.rust-lang.org/nightly/rustc/check-cfg.html), have an addendum that this can also be set in `build.rs` (link to the directive's docs) and lint config (show a short example)

Reply:
> Update the [lint docs](https://doc.rust-lang.org/nightly/rustc/lints/listing/warn-by-default.html#unexpected-cfgs) to point to the [check-cfg docs](https://doc.rust-lang.org/nightly/rustc/check-cfg.html) so users can discover how to customize things

That is indeed something we should do. I will do that in this PR.

> Soon after It has this basic form: in the [check-cfg docs](https://doc.rust-lang.org/nightly/rustc/check-cfg.html), have an addendum that this can also be set in build.rs (link to the directive's docs) and lint config (show a short example)

I would rather prefer if we not do that. That section is specifically for the `--check-cfg` flag and adding in the middle of it some cargo knowledge seems out place, in particular since Cargo is not the build system that uses `--check-cfg`.

In general I would really prefer to keep the current `--check-cfg` doc as is, that's why I propose to add a small "sub-page" dedicated to those Cargo specificity so that 1. we can easily link to it and 2. so that users of Cargo are not overwhelm with all the technicalities of the `--check-cfg` flag when they only care about Cargo (at first at least).

Reply:
I just updated the lint documentation to refer to the check-cfg docs.